### PR TITLE
readme add arguments

### PR DESCRIPTION
**Pull request by RuoyuWang-MS**
readme add arguments

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -178,6 +178,26 @@ The "Delete Staging Deployment" action allows you to delete the deployment not r
           app-name: <app name>
 ```
 
+## Arguments
+> [!NOTE]
+> Some arguments are only applicable for certain settings of the `action` argument. The Action column below specifies the pertinent actions for each argument. Any argument listed as Required is only required for the pertinent Action(s).
+
+|Argument|Action&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|Description|
+|--- |--- |--- |
+|`action`<br/>Action|all|(Required) The action to be performed by this task.<br/>One of: `deploy`, `set-production`, `delete-staging-deployment`<br/>Default value: `deploy`|
```
Comment (allxiao) [inline README.md:187]: I think we can remove `<br/>XXX` in the `Argument` column. It does not offer much clarity as it's duplicated in the `Description` column.

File: README.md [hunk 2]
```diff
@@ -178,6 +178,26 @@ The "Delete Staging Deployment" action allows you to delete the deployment not r
           app-name: <app name>
 ```
 
+## Arguments
+> [!NOTE]
+> Some arguments are only applicable for certain settings of the `action` argument. The Action column below specifies the pertinent actions for each argument. Any argument listed as Required is only required for the pertinent Action(s).
+
+|Argument|Action&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;|Description|
```
Comment (allxiao) [inline README.md:185]: Remove `&nbsp;` s

File: README.md [hunk 3]
```diff
@@ -178,6 +178,26 @@ The "Delete Staging Deployment" action allows you to delete the deployment not r
...
+|--- |--- |--- |
+|`action`<br/>Action|all|(Required) The action to be performed by this task.<br/>One of: `deploy`, `set-production`, `delete-staging-deployment`<br/>Default value: `deploy`|
+|`azure-subscription`<br/>Azure Subscription ID|all|(Required) The id of the azure subscription.|
```
Comment (allxiao) [inline README.md:188]: Suggest to add an extra column for `Required`, and remove the `(xxx)` in description.

Specifically for this one: `The Azure subscription ID for the target Azure Spring Cloud instance.`

File: README.md [hunk 4]
```diff
@@ -178,6 +178,26 @@ The "Delete Staging Deployment" action allows you to delete the deployment not r
...
+|`service-name`<br/>Service Name|all|(Required) The name of the Azure Spring Cloud service instance.|
+|`app-name`<br/>App Name|all|(Required) The name of the Azure Spring Cloud app to deploy. The app must exist prior to task execution.
+|`use-staging-deployment`<br/>Use Staging Deployment|deploy<br/>set-production|(Optional) If set to `true`, apply the task to whichever deployment is set as the staging deployment at time of execution. If set to `false`, apply the task to the production deployment.<br/>Default value: `true`|
+|`deployment-name`<br/>Deployment Name |deploy<br/>set-production|(Optional) The name of the deployment to which the action will apply. It has priority over `use-staging-deployment`.|
```
Comment (allxiao) [inline README.md:192]: It overrides the setting of `use-staging-deployment`.

File: README.md [hunk 5]
```diff
@@ -178,6 +178,26 @@ The "Delete Staging Deployment" action allows you to delete the deployment not r
...
+|`create-new-deployment`<br/>Create new deployment |deploy|(Optional) If set to `true` and the deployment specified by `deployment-name` does not exist at execution time, it will be created.<br/>Default value: `false`|
+|`package`<br/>Package or folder|deploy|(Required) The file path to the package containing the application to be deployed (`.jar` file for Java, `.zip` for .NET Core) or to a folder containing the application source to be built. <br/>Default value: ```${{ github.workspace }}/**/*.jar```|
+|`runtime-version`<br/>Runtime Version|deploy|(Optional) The runtime stack for the application.<br/>One of: `Java_8`, `Java_11`, `NetCore_31`,<br/>Default value: `Java_11`|
+|`environment-variables`<br/>Environment Variables|deploy|(Optional) Environment variables to be entered using the syntax &#39;-key value&#39;. Values containing spaces should be enclosed in double quotes. <br/>Example: ```-CUSTOMER_NAME Contoso -WEBSITE_TIME_ZONE "Eastern Standard Time"```|
+|`jvm-options`<br/>JVM Options|deploy|(Optional) A string containing JVM Options. <br/> Example: `-XX:+UseG1GC -XX:MaxRAMPercentage=70 -Dazure.keyvault.enabled=true -Dazure.keyvault.uri=https://myvault.vault.azure.net/`|
```
Comment (allxiao) [inline README.md:197]: Lifecycle already applied `-XX:MaxRAMPercentage=70`, will use get a conflict if they apply this configuration?

**Review by allxiao — CHANGES_REQUESTED** [review]
added some comments.

File: README.md (hunk 2 above)
Comment (RuoyuWang-MS) [reply README.md:185]: Removed, but the Action column will wrap at set-production

File: README.md (hunk 5 above)
Comment (RuoyuWang-MS) [reply README.md:197]: I am not quite sure about the case. This is the example of pipeline task. Need I replace with sth like `-Xms2048m -Xmx2048m -Dspring.profiles.active=mysql`

File: README.md [hunk 6]
```diff
@@ -178,6 +178,26 @@ The "Delete Staging Deployment" action allows you to delete the deployment not r
...
+|`service-name`|all|Required| The name of the Azure Spring Cloud service instance.|
+|`app-name`|all|Required| The name of the Azure Spring Cloud app to deploy. The app must exist prior to task execution.
+|`use-staging-deployment`|deploy<br/>set-production|Optional| If set to `true`, apply the task to whichever deployment is set as the staging deployment at time of execution. If set to `false`, apply the task to the production deployment.<br/>Default value: `true`|
+|`deployment-name`|deploy<br/>set-production|Optional| The name of the deployment to which the action will apply. It overrides the setting of `use-staging-deployment`.|
+|`create-new-deployment`|deploy|Optional| If set to `true` and the deployment specified by `deployment-name` does not exist at execution time, it will be created.<br/>Default value: `false`|
+|`package`|deploy|Required| The file path to the package containing the application to be deployed (`.jar` file for Java, `.zip` for .NET Core) or to a folder containing the application source to be built. <br/>Default value: ```${{ github.workspace }}/**/*.jar```|
```
Comment (yucwan) [inline README.md:194]: please move this up to after app-name so that all the required are listed first. others lgtm

**Review by yucwan — APPROVED** [review]

File: README.md (hunk 6 above)
Comment (RuoyuWang-MS) [reply README.md:194]: As the discription above the table, this is not required of all actions. This only required in deploy action. The order follows pipeline task doc. So I am not sure if it should be placed after app-name.

**Review by allxiao — APPROVED** [review]

**Pull request merged**